### PR TITLE
[14.0][FW] queue_job: Forward updates from 13.0

### PR DESCRIPTION
**Pull request by dzungtran89**
Porting PRs:
- #303
- #305
- #302
- #310 
- #309 
- #312 
- #328 
- #352 
- #356

### CODE REVIEW

File: queue_job/migrations/13.0.3.7.0/pre-migration.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html)
+
+import logging
+
+from odoo.tools.sql import column_exists
+
+_logger = logging.getLogger(__name__)
+
+
+def migrate(cr, version):
```
Comment (simahawk) [inline queue_job/migrations/13.0.3.7.0/pre-migration.py]: bump version manual and merge these scripts to run in one version for v14

**Review by simahawk — CHANGES_REQUESTED** [review]

File: queue_job/migrations/13.0.3.7.0/pre-migration.py (hunk 1 above)
Comment (dzungtran89) [reply queue_job/migrations/13.0.3.7.0/pre-migration.py]: ok, thank you. I'll update the script and have a fix for the failed travis

Comment (dzungtran89) [reply queue_job/migrations/13.0.3.7.0/pre-migration.py]: @simahawk the PR has been updated

File: queue_job/__manifest__.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 {
     "name": "Job Queue",
-    "version": "14.0.1.3.2",
+    "version": "14.0.1.3.3",
```
Comment (simahawk) [inline queue_job/__manifest__.py]: this is more than a patch

**Review by simahawk — APPROVED** [review]
version number looks wrong for such a big change.
Beside this, LG (code review only).
Thanks!

File: queue_job/__manifest__.py (hunk 2 above)
Comment (dzungtran89) [reply queue_job/__manifest__.py]: good day @simahawk , so bumping the minor would be better `14.0.1.4.0`

Comment (simahawk) [reply queue_job/__manifest__.py]: yes, thanks

**Review by sebalix — APPROVED** [review]

**Comment by simahawk** [comment]
@dzungtran89 could you rebase?

**Comment by hailangvn** [comment]
Good day @simahawk, this PR has been rebased.

**Pull request merged**